File: typescript/session3/td/ConsRec.ts
import { Mot } from "./Mot"

export class ConsRec extends Mot {
    private var_lettre : string;
    private var_reste : Mot;
    private var_taille : number;

    public constructor(lettre : string, reste : Mot) {
        super();
        this.var_lettre = lettre;
        this.var_reste = reste;
        this.var_taille = 1 + reste.taille();
    }

    public casCons() : boolean {
        return true;
    }

    public lettre() : string {
        return this.var_lettre;
    }

    public reste() : Mot {
        return this.var_reste;
    }

    public taille() : number {
        return this.var_taille;
    }
}

File: java/cours/src/session3/td/src/ConsRec.java
public class ConsRec implements Mot {
    private char lettre;
    private Mot reste;

    ConsRec(char lettre, Mot reste) {
        this.lettre = lettre;
        this.reste = reste;
    }

    @Override
    public char lettre() {
        return lettre;
    }

    @Override
    public Mot reste() {
        return reste;
    }

    @Override
    public int taille() {
        return 1 + reste.taille();
    }

    @Override
    public boolean casCons() {
        return true;
    }

    public String toString() {
        return "" + lettre + reste.toString();
    }
}
